express-notes-api/index.js

```diff
@@ -0,0 +1,151 @@
+const express = require('express');
+const app = express();
+
+
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/notes', (req, res) => {
+  const notesArr = [];
+  const datajson = require('./data.json');
+  const notes = datajson.notes
+  for (const id in notes) {
+    notesArr.push(notes[id]);
+  }
+  res.status(200).json(notesArr);
+})
+
+app.get('/api/notes/:id', (req, res) => {
+  const datajson = require('./data.json');
+  const noteId = req.params.id
+  const notes = datajson.notes
+  const note = notes[noteId];
+  if (`${parseInt(noteId)}` === 'NaN' || parseInt(noteId) < 1) {
```

Comment:
nothing is every === to NaN, you should be checking if the number is not an integer here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
check out the isInteger method

Reply:
Yah I figured out in a previous challenge that nothing is equal to NaN that's why if it is NaN it comes out as a string "NaN" and I am checking if it is equal to that string. It works, but if you think isInteger is better I can change it.